Spatial index and data type docs (#1018)
memgraph/documentation: pages/fundamentals/indexes.mdx

</Callout>

### Spatial index

To run queries such as:
```cypher
WITH point({x:1, y:1}) AS p
MATCH (node:Label) WHERE point.distance(p, node.property) < 1
RETURN node;
```

> in a performant way, feel free to create a new point index as follows:

Reviewer: Are we using point index or spatial index in text?

</Callout>

### Point index

To run queries such as:
```cypher
WITH point({x:1, y:1}) AS p
MATCH (node:Label) WHERE point.distance(p, node.property) < 1
RETURN node;
```
in a performant way, feel free to create a new point index as follows:
```cypher
CREATE POINT INDEX ON :Label(property);
```

<Callout type="info">

> In Memgraph version 2.21 to utilize the point index in queries using point.distance the comparison point
> needs to exist as its own variable
> For example instead of:

Reviewer: ```suggestion
From Memgraph 2.21, to utilize the point index in queries using the point.distance function, the comparison point
needs to exist as a variable first. For example, instead of:
```
